test(browser) : enhance test coverage for http and auth params utilities

KavishkaVenuka opened the pull request:
- Add comprehensive test cases for hasAuthParamsInUrl utility
- Add extensive test coverage for http utility including:
  - Request method testing with various HTTP methods
  - RequestAll method testing for batch operations
  - Edge case handling
  - Error scenarios
  - Integration tests
  - Concurrent request handling

This change improves the test coverage and reliability of the browser package's utilities.

### Purpose
-This PR enhances the test coverage for utilities in the browser package by adding comprehensive test suites for both hasAuthParamsInUrl and HTTP utilities. The changes focus on improving code reliability and ensuring robust error handling.

###Key Improvements:
-Enhanced test coverage for hasAuthParamsInUrl utility

###URL parameter validation
-Edge case handling
-Special character testing
-Parameter positioning scenarios

###Comprehensive HTTP utility test coverage
-Various HTTP method validations (GET, POST, PUT, DELETE)
-Batch operation testing via RequestAll method
-Concurrent request handling scenarios
-Error case validation
-Integration test scenarios
-Edge case handling for various request states

###Related Issues
-N/A

###Related PRs
-N/A

###Checklist
 -Followed the [CONTRIBUTING] guidelines
 -Manual test round performed and verified
 -Unit tests provided in:
---->[hasAuthParamsInURL.test.ts]
---->packages/browser/src/utils/__tests__/http.test.ts
 -Documentation provided (No documentation updates required as this is test-only changes)

###Security checks
 -Followed secure coding standards in [http://wso2.com/technical-reports/wso2-secure-engineering-guidelines]
 -Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets

###Additional Notes
-All test cases follow the project's testing conventions
-Tests are designed to be maintainable and readable
-No production code changes were made, only test improvements

KavishkaVenuka committed:
test(browser): enhance test coverage for http and auth params utilities

- Add comprehensive test cases for hasAuthParamsInUrl utility
- Add extensive test coverage for http utility including:
  - Request method testing with various HTTP methods
  - RequestAll method testing for batch operations
  - Edge case handling
  - Error scenarios
  - Integration tests
  - Concurrent request handling

This change improves the test coverage and reliability of the browser package's utilities.

KavishkaVenuka committed:
Merge branch 'main' into test/asgardeo-browser#166

KavishkaVenuka committed:
test(browser): enhance test coverage for theme utilities

- Add comprehensive test suite for getActiveTheme utility
- Implement test cases for:
  - Explicit theme mode detection
  - System theme preference handling
  - Class-based theme detection
  - Edge case scenarios
  - Theme mode fallbacks

- Improve test organization with proper describe blocks
- Add proper setup/teardown for window.matchMedia mocks
- Include tests for all theme modes: light, dark, system, and class-based

KavishkaVenuka committed:
test(browser): enhance config type validation and test coverage

- Add comprehensive test cases for AsgardeoBrowserConfig type
- Implement storage type validation tests
- Add required and optional field validation
- Improve test organization with proper describe blocks
- Add runtime validation for invalid storage types
- Use type-safe assertions and constants

KavishkaVenuka committed:
chore(browser): add unit tests for AsgardeoBrowserClient to increase coverage

Add tests for the default export and subclassing behavior of AsgardeoBrowserClient:
- verify default export is a class
- ensure a concrete subclass can be instantiated
- assert inheritance from AsgardeoJavaScriptClient
- validate initialize and getConfiguration behavior
- cover core method stubs: isLoading, isSignedIn, getUser, getUserProfile, organizations APIs, getAccessToken, exchangeToken
- exercise signIn/signOut/signUp overloads with stubbed returns

Tests are colocated under src/__tests__/AsgardeoBrowserClient.test.ts to match package conventions

Refs: @asgardeo/browser unit test coverage task

KavishkaVenuka committed:
Merge branch 'asgardeo:main' into main

brionmario commented on packages/browser/jest.config.js:
Hey @KavishkaVenuka,

Could you please explain why we need a Jest Config?

KavishkaVenuka replied:
My bad sir. I'll remove it.

brionmario commented on packages/browser/package.json:
```diff
@@ -53,7 +53,7 @@
     "prettier": "^2.6.2",
     "rimraf": "^6.0.1",
-    "typescript": "~5.7.2",
+    "typescript": "~5.7.3",
```
Any reason for this version bump?

KavishkaVenuka committed:
Remove unnecessary Jest config as per review feedback

KavishkaVenuka committed:
Merge branch 'asgardeo:main' into main

KavishkaVenuka commented on packages/browser/package.json:
```diff
@@ -74,4 +74,4 @@
     "access": "public"
   }
-}
+}
```
Done sir.

KavishkaVenuka committed:
chore(browser): resolve eslint violations and apply prettier to tests
add and normalize license headers in test files
add missing type annotations to satisfy @typescript-eslint/typedef
reorder object keys to comply with sort-keys in mocks and assertions
adjust matchMedia mocks and expectation objects to meet lint rules
normalize line endings to LF to satisfy prettier/prettier on windows
add targeted eslint disables for test-only patterns (class-methods-use-this, no-underscore-dangle, typedef where necessary)
recreate placeholder utils test file with correct header and local disables
format all browser test files using package prettier config

KavishkaVenuka committed:
chore(browser): update test tooling and lockfile; fix typings in config tests
remove @types/jest from lockfile and align Vitest-related entries
bump/resolve @vitest/browser and @vitest/coverage-v8 to match vitest 3.1.3
add explicit types and safe casting in src/models/__tests__/config.test.ts
keep formatting consistent with repo Prettier config
